### PR TITLE
Api v2 verify test

### PR DESCRIPTION
**Pull request by igorosip0v**
This PR: 
- Improves typing of nullifier in the `canVerifyForAction` function
  - This also removes check for the `null` for the uses field as we made this field not nullable
- Allows action to be `""` in `/v2/verify/` body validation schema
  - @0xPenryn @andy-t-wang Please let me know if this is wrong, I'll revert it and remove test case for this.
  - change is `yup.required` -> `yup.defined`
- Moves /verify endpoint code files from `/web/api/verify/...` to `/web/api/v2/verify/...`
  - I assume there will be more /v2/ endpoints in the future, let's keep endpoint code under v2 too.
- Fixes the way we are getting body in API route handler 
  - `const body = req.json()` instead of `req.body`
- Adds tests for `/v2/verify` endpoint

### CODE REVIEW

File: web/tests/api/v2/verify.test.ts
Comment (0xPenryn) [inline web/tests/api/v2/verify.test.ts:80]: unhashed signal shouldn't be included

```suggestion
const validBody = {
  ...semaphoreProofParamsMock,
  signal: undefined,
  action: "verify",
};
```

File: web/tests/api/v2/verify.test.ts
Comment (0xPenryn) [inline web/tests/api/v2/verify.test.ts:1]: we should add test cases for invalid semaphore proofs, for no signalHash provided (valid when `signal: ""` in IDKit), and for improper signalHash format (should be hex string of 66 characters, inclusive of `0x`)

File: web/api/v2/verify/index.ts
Comment (0xPenryn) [inline web/api/v2/verify/index.ts:22]: I think I'd prefer to keep this as `.required()` to prevent proof overlap between Incognito Actions and Sign In with World ID, pending approval from @andy-t-wang -- is there any way for us to tell if there are apps using `action: ""` for incognito actions?

If we do move forward with preventing empty actions, we should also update IDKit to make `action` a required parameter.

**Review by 0xPenryn — CHANGES_REQUESTED** [review]
looks great for the most part, but there are a few more test cases I think we should have and we need a final decision on how to handle `action: ""`

**Review by 0xPenryn — CHANGES_REQUESTED** [review]
looks great for the most part, but there are a few more test cases I think we should have and we need a final decision on how to handle `action: ""`

File: web/api/v2/verify/index.ts
Comment (igorosip0v) [reply web/api/v2/verify/index.ts]: Updated.

File: web/tests/api/v2/verify.test.ts
Comment (igorosip0v) [reply web/tests/api/v2/verify.test.ts]: Updated

File: web/tests/api/v2/verify.test.ts
Comment (igorosip0v) [reply web/tests/api/v2/verify.test.ts:1]: @0xPenryn I've added test cases, but not quite sure about this one: 
> for no signalHash provided

In yup schema it simply has default value that will be applied if there is no signal_hash.

**Review by 0xPenryn — APPROVED** [review]

**Pull request merged**
